File: sdks/ruby/lib/novabilling/types/login_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class LoginResponse < Internal::Types::Model
      field :access_token, -> { String }, optional: false, nullable: false, api_name: "accessToken"
      field :refresh_token, -> { String }, optional: false, nullable: false, api_name: "refreshToken"
      field :tenant, -> { Novabilling::Types::TenantInfoResponse }, optional: false, nullable: false
    end
  end
end

File: sdks/php/src/Types/LoginResponse.php
<?php

namespace NovaBilling\Types;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class LoginResponse extends JsonSerializableType
{
    /**
     * @var string $accessToken
     */
    #[JsonProperty('accessToken')]
    public string $accessToken;

    /**
     * @var string $refreshToken
     */
    #[JsonProperty('refreshToken')]
    public string $refreshToken;

    /**
     * @var TenantInfoResponse $tenant
     */
    #[JsonProperty('tenant')]
    public TenantInfoResponse $tenant;

    /**
     * @param array{
     *   accessToken: string,
     *   refreshToken: string,
     *   tenant: TenantInfoResponse,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->accessToken = $values['accessToken'];
        $this->refreshToken = $values['refreshToken'];
        $this->tenant = $values['tenant'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
